lib/src/impl/realtime/channels.dart

```diff
@@ -136,14 +136,29 @@ class RealtimePlatformChannel extends PlatformObject implements spec.RealtimeCha
   }
 
   @override
-  Future<void> subscribe({String event, List<String> events, spec.EventListener<spec.Message> listener}) {
-    // TODO: implement subscribe
-    return null;
-  }
-
-  @override
-  void unsubscribe({String event, List<String> events, spec.EventListener<spec.Message> listener}) {
-    // TODO: implement unsubscribe
+  Stream<spec.Message> subscribe({
+    String name,
+    List<String> names
+  }) {
+    Stream<spec.Message> stream = listen(PlatformMethod.onRealtimeChannelMessage, _payload).transform<spec.Message>(
+      StreamTransformer.fromHandlers(
+        handleData: (dynamic value, EventSink<spec.Message> sink){
+          spec.Message message = value as spec.Message;
+          if (names!=null){
+            if(names.contains(message.name)){
+              sink.add(message);
+            }
+          } else if (name!=null) {
+            if(message.name==name){
+              sink.add(message);
+            }
+          } else {
+            sink.add(message);
+          }
+        }
+      )
+    );
+    return stream;
```

Comment:
It's not clear to me what `StreamTransformer.fromHandlers(...)` is required for.
My suggestion would be
```dart
  Stream<spec.Message> subscribe({String name, List<String> names}) {
    final subscribedNames =
        {if (name != null) name, ...?names}.where((n) => n != null).toList();
    return listen(PlatformMethod.onRealtimeChannelMessage, _payload)
        .map((message) => message as spec.Message)
        .where((message) =>
            subscribedNames.isEmpty ||
            subscribedNames.any((n) => n == message.name));
  }
```

Reply:
Also, https://github.com/ably/ably-flutter/pull/20#discussion_r464451601

`on` method in `realtime/channels.dart` and `realtime/connection` to be optimized in same fashion

Reply:
Addressed 5359c91a64e4da5d7988a18d022be5e22038013f